src/ripple/rpc/impl/DeliveredAmount.cpp

```diff
@@ -71,12 +81,12 @@ insertDeliveredAmount(
         // Ledger 4594095 is the first ledger in which the DeliveredAmount field
         // was present when a partial payment was made and its absence indicates
         // that the amount delivered is listed in the Amount field.
-        bool const ledgerIndexCheck = ledgerIndex >= 4594095;
+        bool const ledgerIndexCheck = getLedgerIndex() >= 4594095;
         // If the ledger closed long after the DeliveredAmount code was deployed
         // then its absence indicates that the amount delivered is listed in the
         // Amount field. DeliveredAmount went live January 24, 2014.
         // 446000000 is in Feb 2014, well after DeliveredAmount went live
-        bool const closeTimeCheck = closeTime > NetClock::time_point{446000000s};
+        bool const closeTimeCheck = getCloseTime() > NetClock::time_point{446000000s};
```

Comment:
`getCloseTime` can be expensive as it hits the node store and possibly SQLite. I recommend evaluating `closeTimeCheck`  only if `ledgerIndexCheck` is none.

Reply:
@miguelportilla Agreed. The whole point of passing in the lambda was to evaluate it only when needed, and then I evaluated here unnesisarily anyway. Good catch. Fixed in: `575212aa1 * delivered_amt [FOLD] Do not evaluate `getCloseTime` unless needed`